common/src/main/java/com/wynntils/wc/custom/item/WynnItemStack.java

```diff
@@ -35,19 +36,37 @@ public List<Component> getOriginalTooltip() {
     }
 
     public void addProperty(ItemProperty property) {
-        if (getProperty(property.getClass()) != null) return; // don't allow duplicate properties of same type
+        if (hasProperty(property.getClass())) return; // don't allow duplicate properties
         this.properties.add(property);
     }
 
-    public <T extends ItemProperty> T getProperty(Class<T> propertyType) {
+    /**
+     * Returns the specified property or the first property of the specified type, if it exists.
+     * Otherwise, returns null.
+     */
+    public <T> T getProperty(Class<T> propertyType) {
         for (ItemProperty property : properties) {
-            if (property.getClass().equals(propertyType)) return (T) property;
+            if (propertyType.isAssignableFrom(property.getClass())) return (T) property;
         }
         return null; // no match
     }
 
-    public boolean hasProperty(Class<? extends ItemProperty> propertyType) {
+    /**
+     * Returns true if the specified property, or a property of the specified type, is present
+     */
+    public boolean hasProperty(Class<?> propertyType) {
         // getProperty returns null if no property of the given type is present
         return (getProperty(propertyType) != null);
     }
+
+    /**
+     * Returns all the present properties of the specified type.
+     */
+    public <T extends PropertyType> List<T> getProperties(Class<T> propertyType) {
```

Comment:
How would getProperties be used? Would it be used to deal with potential "conflicts" in `PropertyType`s with the same functionality, or am I just confused

Reply:
the intended use is for types that don't necessarily conflict with themselves, such as text overlays. if an item has multiple text overlays on it, there needs to be a way to collect and render all of them